firebase-query.html

```diff
@@ -331,6 +334,9 @@
             this.async(function() {
               this.syncToMemory(function() {
                 this.splice('data', this.__indexFromKey(key), 1);
+                if (this.data.length === 0) {
```

Comment:
Doesn't this imply the data in memory does not reflect what is in Firebase?

If that is the case then this isn't true for `[]` which could very well exist in either location and be valid.